code/math/vecmat.cpp

```diff
@@ -2954,6 +2954,46 @@ void vm_interpolate_angles_quick(angles *dest0, angles *src0, angles *src1, floa
 	dest0->b = src0->b + (arc_measures.b * interp_perc);
 }
 
+// Interpolate between two matrices, using t as a percentage progress between them.
+// derived by Asteroth from our AI code
+void vm_interpolate_matrices(matrix* out_orient, matrix* curr_orient, matrix* goal_orient, float t) 
+{
+	// check the two cases where it doesn't make sense to go through the whole function
```

Comment:
This is only one case?  Unless this is also meant to refer to the check on line 2982